analysis/analysis/ic50.py

```diff
@@ -0,0 +1,65 @@
+import json
+import numpy as np
+from scipy.optimize import curve_fit
+import sys
+from sklearn.preprocessing import MinMaxScaler
+
+
+def find_nearest_idx(array, value):
+    array = np.asarray(array)
+    idx = (np.abs(array - value)).argmin()
+    return idx
+
+
+def sigmoid(x, a, b, c, d):
+    return ((a - d) / (1.0 + ((x / c) ** b))) + d
+
+
+def replace_duplicates(measurements):
+    unique_concentrations = [c for c in {s[0] for s in measurements} if not np.isnan(c)]
+
+    unique_measurements = []
+    for c in unique_concentrations:
+        specific_measurements = measurements[measurements[:, 0] == c]
+        specific_signals = [s[1] for s in specific_measurements]
+        unique_measurements.append([c, np.mean(specific_signals)])
+
+    unique_measurements = np.array(unique_measurements)
+    sorted_measurements = unique_measurements[unique_measurements[:, 0].argsort()]
+    return np.array([s[0] for s in sorted_measurements]), np.array(
+        [s[1] for s in sorted_measurements])
+
+
+def compute_sigmoid(measurements):
+    concentrations, signals = replace_duplicates(measurements)
+
+    scaler = MinMaxScaler()
+    signals_scalered = scaler.fit_transform(signals.reshape(-1, 1))
+    signals_scalered = signals_scalered.reshape(-1)
+
```

Comment:
You can silence warnings, so that the logs will be empty. Something like this:
`import warnings`
`warnings.filterwarnings("ignore")`

Reply:
Hm, as I saw warnings are in stderr, and I think we should show all warnings and errors to catch unexpected behaviour